compiler/rustc_hir_typeck/src/pat.rs

```diff
@@ -436,7 +436,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             // ```
             //
             // See issue #46688.
-            PatKind::Ref(inner, mutbl) => AdjustMode::ResetAndConsumeRef(*mutbl, inner.span),
+            PatKind::Ref(inner, mutbl) => AdjustMode::ResetAndConsumeRef(*mutbl, pat.span.until(inner.span.find_ancestor_inside(pat.span).unwrap())),
```

Comment:
The documentation for `find_ancestor_inside` explicitly says to not use it to expand the span, which I understood would probably apply to `until` too. Did `find_ancestor_inside_same_ctxt` not do what you want?

Reply:
I am shrinking the span, not expanding it?

Reply:
`until` doesn't know that though. I don't really understand what `SyntaxContext`s are but it sounded good to use the same one?

Reply:
Looking at the `rustc_span` source code, I think it's fine, because `until` will use the `SyntaxContext` of the first (outer) span anyway—so anything weird about the `SyntaxContext` of the span passed as the last argument to `until` won't influence the result. (IOW, we are not "extending the *returned* span", we are shrinking a different span that is not the returned span.)

Reply:
Ok, that's fine then